[macOS] Fix TestWebKitAPI.WebKit.ResponsivenessTimerCrash timeout

salinas-miguel opened the pull request:
#### 857e02522abcceb9cb46dd2f9ac2e79a0a899c39
<pre>
[macOS] Fix TestWebKitAPI.WebKit.ResponsivenessTimerCrash timeout
<a href="https://bugs.webkit.org/show_bug.cgi?id=253660">https://bugs.webkit.org/show_bug.cgi?id=253660</a>
rdar://problem/106509558

Reviewed by Chris Dumez.

The ProcessThrottler is effectively disabled until didConnectToProcess()
is called. m_processIdentifier is 0 until didConnectToProcess(). Some OS
versions never call didConnectToProcess(). We should not return true for
isSuspended() when the ProcessThrottler is effectively disabled.

* Source/WebKit/UIProcess/ProcessThrottler.h:
(WebKit::ProcessThrottler::isSuspended const):

Canonical link: <a href="https://commits.webkit.org/262381@main">https://commits.webkit.org/262381@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/81eab37d56bd4c783f725aff346df080494c133d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/509 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/525 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/550 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/569 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/454 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/572 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/611 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/699 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/517 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/497 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/488 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/571 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/552 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/477 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/526 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/452 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/496 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/483 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/501 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/444 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/490 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/344 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/491 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

cdumez reviewed (approved):
Code change looks correct.

salinas-miguel committed:
[macOS] Fix TestWebKitAPI.WebKit.ResponsivenessTimerCrash timeout
https://bugs.webkit.org/show_bug.cgi?id=253660
rdar://problem/106509558

Reviewed by Chris Dumez.

The ProcessThrottler is effectively disabled until didConnectToProcess()
is called. m_processIdentifier is 0 until didConnectToProcess(). Some OS
versions never call didConnectToProcess(). We should not return true for
isSuspended() when the ProcessThrottler is effectively disabled.

* Source/WebKit/UIProcess/ProcessThrottler.h:
(WebKit::ProcessThrottler::isSuspended const):

Canonical link: https://commits.webkit.org/262381@main
